subiquitycore/utils.py

```diff
@@ -176,3 +176,10 @@ def disable_subiquity():
                  "snap.subiquity.subiquity-service.service",
                  "serial-subiquity@*.service"])
     return
+
+
+def matching_dicts(items, **kw):
+    """Given an input sequence of dictionaries, return a list of dicts where
+    the supplied keyword arguments all match those items."""
+    return [item for item in items
+            if all(item.get(k) == v for k, v in kw.items())]
```

Comment:
Note that because `.get()` returning `None` when key is not found, any filter that expects a `None` value will essentially be ignored:

```python
print(matching_dicts([{"foo", "bar"}], baz=None))
[{'foo': 'bar'}]
```
I would tend to think something like the below is better, but this means we change the existing implementation of `match`, so I don't know if it's a good thing. Maybe a TODO is better until we figure out?

```python
return [item for item in items
        if all(k in item and item[k] == v for k, v in kw.items())]

Reply:
I'd rather implement this now, I see it as a strictly better version.
If this had regressed an API test (doesn't seem to) I would have considered that a bug in the respective API test.

Reply:
Perfect, thanks!